stdlib/SparseArrays/src/linalg.jl

```diff
@@ -1028,9 +1028,16 @@ function opnorm(A::SparseMatrixCSC, p::Real=2)
     m, n = size(A)
     if m == 0 || n == 0 || isempty(A)
         return float(real(zero(eltype(A))))
-    elseif m == 1 || n == 1
-        # TODO: compute more efficiently using A.nzval directly
-        return opnorm(Array(A), p)
+    elseif m == 1
+        if p == 1
+            return norm(A.nzval, Inf)
+        elseif p == 2
+            return norm(A.nzval, 2)
+        elseif p == Inf
+            return norm(A.nzval, 1)
+        end
```

Comment:
If I'm not mistaken, you now need an else-branch which throws the "wrong p" error?

Reply:
@dkarrasch It falls through the `if` block and reaches the last line of the method
```julia
throw(ArgumentError("invalid operator p-norm p=$p. Valid: 1, Inf"))
```
which produces the error message for the *wrong* `p`.